packages/plugin-node-modules/sources/NodeModulesLinker.ts

```diff
@@ -200,23 +240,37 @@ async function writeInstallState(project: Project, locatorMap: NodeModulesLocato
   });
 };
 
-async function findInstallState(project: Project, {unrollAliases = false}: {unrollAliases?: boolean} = {}): Promise<NodeModulesLocatorMap | null> {
+async function findInstallState(project: Project, {unrollAliases = false}: {unrollAliases?: boolean} = {}): Promise<InstallState | null> {
   const rootPath = project.cwd;
   const installStatePath = ppath.join(rootPath, NODE_MODULES, INSTALL_STATE_FILE);
 
   if (!xfs.existsSync(installStatePath))
     return null;
 
   const locatorState = parseSyml(await xfs.readFilePromise(installStatePath, `utf8`));
-  delete locatorState.__metadata;
-
   const locatorMap: NodeModulesLocatorMap = new Map();
+  const binSymlinks: BinSymlinkMap = new Map();
+
+  if (locatorState.__metadata.version > STATE_FILE_VERSION)
```

Comment:
Shouldn't it be the opposite? If we have a higher version that the serialized one, ignore the previous state?

Reply:
I think this `if` does exactly what you say, if we have a higher version than serialized one - ignore previous state (return empty maps)

Reply:
I'm still not sure - `locatorState` comes from the serialized file, and `STATE_FILE_VERSION` is the constant in the current binary. So I read the condition as "if the serialized version is higher than the local one, return an empty locator map and bin list". Am I missing something?

Reply:
If the serialized version is higher than the plugin supports - we return empty state - yes. It is for the case when the state file was generated by the newer plugin version and then older plugin reads new state - it should discard new state, because it doesn't know how to properly handle it

Reply:
Even like this, shouldn't it be `locatorState.__metadata.version !== STATE_FILE_VERSION`? Otherwise a new Yarn will use the data from the old one 🤔

Reply:
Yes, new Yarn will upgrade old data and use it